packages/mui-material/src/Slider/Slider.d.ts

```diff
@@ -296,6 +301,24 @@ export declare const SliderTrack: React.FC<SliderTrackProps>;
 export declare const SliderThumb: React.FC<SliderThumbProps>;
 export declare const SliderValueLabel: React.FC<SliderValueLabelProps>;
 
+interface SliderType {
+  <RootComponent extends React.ElementType, Value extends number | number[]>(
+    props: {
+      /**
+       * The component used for the root node.
+       * Either a string to use a HTML element or a component.
+       */
+      component: RootComponent;
+      value?: Value;
+    } & OverrideProps<SliderTypeMap<SliderTypeMap['defaultComponent'], {}, Value>, RootComponent>,
+  ): JSX.Element | null;
+  <Value extends number | number[]>(
+    props: { value?: Value } & DefaultComponentProps<
```

Comment:
Could you explain why it's necessary to add value explicitly here? Isn't it enough that it's in `SliderTypeMap`?

Reply:
My question was, why is it necessary to extract the value if it's already in the `SliderTypeMap` 🤔 

```ts
props:
  { value?: Value } // <--- why is this necessary
  & DefaultComponentProps<
    SliderTypeMap<SliderTypeMap['defaultComponent'], {}, Value> // <--- if it's already passed here
  >
```

I don't see that done in the Typescript playground you shared, but I may be missing something.

While reviewing the `NoInfer` issue, I found [this comment](https://github.com/microsoft/TypeScript/issues/14829#issuecomment-288902999). Would that work for us? I think it's better as it [does not rely on the deferral of evaluation](https://github.com/microsoft/TypeScript/issues/14829#issuecomment-1457829132). I've tried it in [this playground](https://www.typescriptlang.org/play?#code/PTAEDcFMCcGcEsD2A7U9UFsCGsAuNQAjaLZAYwAsAoAMwFdzclUBlAG3gBMYAeANSxs6kUJAAe+ZJ1ihkdDIRgAfOQpgBtALoAaUAHlkAYQqkA5pAFCR4ydNCXhAPgAUKY2cgAuZ+EHDP+kYmyOYOkACUoAC8jhCIXNq+Vp5h4QDeVFQAvpnsXDDOPn5esvKK0JExoGlZ2uoATJrhVHnc0IVJ-qVq0FqVsTXa9c1UQA) and seems to be working.

Reply:
Regarding adding the `onChange` description twice, is that not happening for `onChangeCommitted`? Is that different somehow?

@michaldudak do you know why this might be happening?

Reply:
I'm not so deep into the docs infra. Perhaps someone from the @mui/docs-infra could help here.

Reply:
Seems to not be the only issue we have with props documentation. It look similar to #38459 where some default value are ignored.

Could you open an issue with the diff to do in order to remove the hack such that we can try a deeper look later?